concent_api/core/payments/bankster.py

```diff
@@ -400,26 +346,31 @@ def settle_overdue_acceptances(
         # subtask_results_accepted_list.
         youngest_payment_ts = max(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
 
-        transaction_hash = service.make_force_payment_to_provider(  # pylint: disable=no-value-for-parameter
```

Comment:
Same here, I remember something about keeping `DepositAccount` locked for the whole process.

Reply:
it is not possible to keep it locked throughout the whole process - either this, or we have nested transactions

Reply:
@rwrzesien 
Yes, I originally wanted it done like this and it's what the blueprint says. But we need to change it unfortunately. Just like @dybi said - we can't get rid of transaction nesting otherwise (actually we can, but it would require maintaining two separate database connections; I don't think Django supports that cleanly).

But I think we can do without it. See https://github.com/golemfactory/concent/pull/1041#discussion_r242920641. The downside is that the function is less fool-proof now. It won't give you an error if you use it wrong. I think we can live with that but I'd still prefer to at least have some comments that tell you about this instead of assuming that everyone can figure that out just by reading the code.

Things like this are the reason why the blueprint is still `[WIP]` even though it may seem complete. Before my vacation I did not really have enough time to solve all these tiny edge cases caused by race conditions. It's really hard to get it right in all cases. I spent a lot of time on it already. That's why I told you back then that you're going to have to somehow deal with the edge cases but could not really tell you how. I found a lot of them later in the review. I'm still going to submit that blueprint in a pull request at some point and that will be a good time to compare it with the implementation again.